### PR TITLE
No create permissions dialog

### PR DESCRIPTION
**Pull request by sjorobekov**
Improve UX, by showing on root page Create Workbook / Collection button, even if user doesn't have permission.
It will open dialog with a prompt that user doesn't have enough permissions.

### CODE REVIEW

File: src/ui/units/collections/components/CollectionPage/hooks/useLayout.tsx [hunk 1]
```diff
@@ -200,27 +238,7 @@ export const useLayout = ({
                                     );
                                 }
                             }}
-                            onCreateCollectionClick={() => {
```
Comment (imsitnikov) [inline src/ui/units/collections/components/CollectionPage/hooks/useLayout.tsx:203]: Is it necessary to create a separate method? It brings duplication of the complex `if`.

**Review by imsitnikov — CHANGES_REQUESTED** [review]

File: src/ui/components/CollectionsStructure/CollectionNoCreatePermissionDialog/CollectionNoCreatePermissionDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import {I18n} from 'i18n';
+import DialogInfo from 'ui/components/DialogInfo/DialogInfo';
+
+import DialogManager from '../../DialogManager/DialogManager';
+import {YfmWrapperContent as YfmWrapper} from '../../YfmWrapper/YfmWrapperContent';
+const i18n = I18n.keyset('component.dialog-collection-no-create-permission.view');
+export const DIALOG_NO_CREATE_COLLECTION_PERMISSION = Symbol('DIALOG_CREATE_WORKBOOK');
+
+export type Props = {
+    message?: React.ReactNode;
```
[the remark this answers is not in the document]
Comment (sjorobekov) [reply src/ui/components/CollectionsStructure/CollectionNoCreatePermissionDialog/CollectionNoCreatePermissionDialog.tsx]: Removed

File: src/ui/units/collections/components/CollectionPage/hooks/useLayout.tsx [hunk 3]
```diff
@@ -173,10 +174,47 @@ export const useLayout = ({
...
+                    props: {
+                        visible: true,
+                        onClose: () => {
+                            dispatch(closeDialog());
+                        },
+                    },
+                }),
+            );
+        }
+    }, [dispatch, isCorrectCollection, isRootCollection, collection, rootCollectionPermissions]);
```
[the remark this answers is not in the document]
Comment (sjorobekov) [reply src/ui/units/collections/components/CollectionPage/hooks/useLayout.tsx]: Updated

File: src/ui/units/collections/components/CollectionPage/hooks/useLayout.tsx [hunk 4]
```diff
@@ -173,10 +174,47 @@ export const useLayout = ({
...
+
     React.useEffect(() => {
         if (
             (isRootCollection && rootCollectionPermissions) ||
-            (isCorrectCollection && collection && collection.permissions)
+            (isCorrectCollection && collection && collection?.permissions)
```
[the remark this answers is not in the document]
Comment (sjorobekov) [reply src/ui/units/collections/components/CollectionPage/hooks/useLayout.tsx]: Yeah, updated. For some reason vscode added it

File: src/ui/units/collections/components/CollectionPage/hooks/useLayout.tsx (hunk 1 above)
Comment (sjorobekov) [reply src/ui/units/collections/components/CollectionPage/hooks/useLayout.tsx:203]: simplified

File: src/ui/components/CollectionsStructure/CollectionNoCreatePermissionDialog/CollectionNoCreatePermissionDialog.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+
+import {I18n} from 'i18n';
+import DialogInfo from 'ui/components/DialogInfo/DialogInfo';
+
+import DialogManager from '../../DialogManager/DialogManager';
+import {YfmWrapperContent as YfmWrapper} from '../../YfmWrapper/YfmWrapperContent';
+const i18n = I18n.keyset('component.dialog-collection-no-create-permission.view');
```
[the remark this answers is not in the document]
Comment (sjorobekov) [reply src/ui/components/CollectionsStructure/CollectionNoCreatePermissionDialog/CollectionNoCreatePermissionDialog.tsx:10]: updated

File: src/ui/units/collections/components/CollectionActions/CollectionActions.tsx [hunk 6]
```diff
@@ -49,17 +49,12 @@ export const CollectionActions = React.memo<Props>(
...
 
-        const showCreateCollection = collection
-            ? collection.permissions?.createCollection
-            : rootCollectionPermissions?.createCollectionInRoot;
+        const showCreateCollection = collection ? collection.permissions?.createCollection : true;
 
-        const showCreateWorkbook = collection
-            ? collection.permissions?.createWorkbook
-            : rootCollectionPermissions?.createWorkbookInRoot;
+        const showCreateWorkbook = collection ? collection.permissions?.createWorkbook : true;
```
Comment (jhoncool) [inline src/ui/units/collections/components/CollectionActions/CollectionActions.tsx:57]: So now we can showAddDemoWorkbook if we can't create workbook

File: src/ui/components/CollectionsStructure/CollectionNoCreatePermissionDialog/CollectionNoCreatePermissionDialog.tsx [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    visible: boolean;
+    onClose: () => void;
+};
+
+export type OpenDialogCollectionNoCreatePermissionArgs = {
+    id: typeof DIALOG_NO_CREATE_COLLECTION_PERMISSION;
+    props: Props;
+};
+
+export const CollectionNoCreatePermissionDialog = React.memo<Props>(({visible, onClose}) => {
```
[the remark this answers is not in the document]
Comment (sjorobekov) [reply src/ui/components/CollectionsStructure/CollectionNoCreatePermissionDialog/CollectionNoCreatePermissionDialog.tsx]: Updated

File: src/ui/units/collections/components/CollectionActions/CollectionActions.tsx (hunk 6 above)
Comment (sjorobekov) [reply src/ui/units/collections/components/CollectionActions/CollectionActions.tsx:57]: Yes, user will see "Demo workbook" button

**Review by jhoncool — DISMISSED** [review]

**Review by jhoncool — APPROVED** [review]

**Pull request merged**